Lib9c/BlockPolicy.cs

```diff
@@ -162,6 +162,16 @@ private InvalidBlockException ValidateMinerAuthority(Block<NCAction> block)
                     miner
                 );
             }
+            
+            // To prevent selfish mining, we define a consensus that blocks with no transactions are do not accepted. 
+            if (block.Transactions.Count <= 0)
+            {
+                return new InvalidMinerException(
+                    $"The block #{block.Index} {block.Hash} (mined by {miner}) should " +
+                    "include at least one transaction.",
+                    miner
+                );
+            }
```

Comment:
Since the 0tx problem is about block, not miner, I think it is right to treat it as a different layer from the auth-miner.

How about checking in the `ValidateNextBlock()` instead?

Reply:
https://github.com/planetarium/lib9c/pull/462/commits/0e33021ec0110fce4406215a3c5d0b5556a6673a accept your idea, I implement `ValidateBlock` and move this check to `ValidateBlock` from `ValidateMinerAuthority`